python/cudaq/handlers/photonics_kernel.py

```diff
@@ -0,0 +1,212 @@
+    """
+    Apply phase shift gate.
+    TBD
+    
+    Args:
+        qudit: An instance of `PyQudit` class.
+        phi: A floating point number for the angle.
+
+    Raises:
+        RuntimeError: If the qudit level is not set.
+        Exception: If input argument is not instance of `PyQudit` class.
+    """
+    _check_args(qudit)
+    cudaq_runtime.photonics.apply_operation('phaseShiftGate', [phi],
+                                            [[qudit.level, qudit.id]])
+
+
+def beam_splitter(q: PyQudit, r: PyQudit, theta: float):
+    """
+    Apply beam splitter gate.
+    TBD
+    
+    Args:
+        q: An instance of `PyQudit` class TBD.
+        r: An instance of `PyQudit` class TBD.
+        theta: A floating point number for the angle.
+    
+    Raises:
+        RuntimeError: If the qudit level is not set.
+        Exception: If input argument is not instance of `PyQudit` class.
+    """
+    _check_args([q, r])
+    cudaq_runtime.photonics.apply_operation('beamSplitterGate', [theta],
+                                            [[q.level, q.id], [r.level, r.id]])
+
+
+def mz(qudits: PyQudit | List[PyQudit], register_name=''):
+    """
+    Measure a single qudit or list of qudits.
+
+    Args:
+        qudits: A single instance or a list of objects of `PyQudit` class.
+    
+    Returns:
+        Measurement results.
+    
+    Raises:
+        RuntimeError: If the qudit level is not set.
+        Exception: If input argument is not instance of `PyQudit` class.
+    """
+    _check_args(qudits)
+    if isinstance(qudits, PyQudit):
+        return cudaq_runtime.photonics.measure(qudits.level, qudits.id,
+                                               register_name)
+    if isinstance(qudits, List):
+        return [
+            cudaq_runtime.photonics.measure(q.level, q.id, register_name)
+            for q in qudits
+        ]
+
+
+class PhotonicsHandler(object):
+    """
+    The `PhotonicsHandler` class serves as to process CUDA-Q kernels for the 
+    `photonics` target.
+    The target must be set to `photonics` prior to invoking a `PhotonicsHandler`.
+
+    The quantum operations in this kernel apply to qudits defined by 
+    `qudit(level=N)` or a list of qudits. The qudits within a kernel must be of
+    the same level.
+
+    Allowed quantum operations are: `plus`, `phase_shift`, `beam_splitter`, and `mz`.
+    """
+
+    def __init__(self, function):
+
+        if 'photonics' != cudaq_runtime.get_target().name:
+            raise RuntimeError(
+                "A photonics kernel can only be used with 'photonics' target.")
+
+        global globalQuditLevel
+        globalQuditLevel = None
+
+        self.kernelFunction = function
+
+        self.kernelFunction.__globals__['qudit'] = qudit
+        self.kernelFunction.__globals__['plus'] = plus
+        self.kernelFunction.__globals__['phase_shift'] = phase_shift
+        self.kernelFunction.__globals__['beam_splitter'] = beam_splitter
```

Comment:
I assume that these would still produce error messages if they were in a kernel that was accidentally targeting a non-photonics backend. Is that correct?

Reply:
Yes, and if I understood correctly, this test showcases that scenario - https://github.com/NVIDIA/cuda-quantum/pull/2055/files#diff-2953518ed100672f01a4c48a36b793f3234dcc0c3f62e7776d9c00b46a9fdff3R152.